playbooks/robusta_playbooks/version_mismatch_enricher.py

```diff
@@ -0,0 +1,100 @@
+import logging
+from typing import Dict, List
+
+from prometrix import PrometheusQueryResult
+from robusta.api import (
+    MarkdownBlock,
+    PrometheusKubernetesAlert,
+    PrometheusParams,
+    SlackAnnotations,
+    TableBlock,
+    action,
+    run_prometheus_query,
+)
+
+
+class VersionMismatchParams(PrometheusParams):
+    """
+    :var version_mismatch_query: The query to get all the build versions.
+    """
+
+    # made as parameter since job label doesn't always exist
+    version_mismatch_query: str = 'count by (git_version, node) (label_replace(kubernetes_build_info{job!~"kube-dns|coredns"}, "git_version", "$1", "git_version", "(v[0-9]*.[0-9]*).*"))'
+
+
+class BuildInfoResults:
+    api_versions: List[str]
+    raw_metrics: List[Dict[str, str]]
+
+    def __init__(self, query_result: PrometheusQueryResult):
+        self.raw_metrics = [result.metric for result in query_result.vector_result]
+        self.api_versions = list(
+            set([metric.get("git_version") for metric in self.raw_metrics if metric.get("node") is None])
+        )
+
+    def is_results_valid(self) -> bool:
+        # there must be atleast two different versions for this alert
+        if len(self.raw_metrics) < 2:
+            logging.error(f"Invalid prometheus results for version_mismatch_enricher.")
+            return False
+
+        # some prometheus's remove specific labels due to space constraints
+        if len([metric for metric in self.raw_metrics if metric.get("node") is not None]) == 0:
+            logging.error(f"Prometheus is missing the 'node' label, unable to detect node versions.")
+            return False
+
+        # you can have multiple versions of the api server in the metrics at once
+        # i.e. at the time of kubernetes upgrade both will show up in the metrics for several minutes
+        if len(self.api_versions) == 0:
```

Comment:
Can we make this more flexible, and support the case we don't have the api server version?
Like, if for example, we have 2 node pools, with different versions